tests/unit/migrations/migrationUtils.test.ts

```diff
@@ -0,0 +1,155 @@
+import { describe, expect, it } from "@jest/globals";
+import { ManagementClient } from "@kontent-ai/management-sdk";
+
+import { Migration } from "../../../src/modules/migrations/models/migration";
+import { MigrationStatus } from "../../../src/modules/migrations/models/status";
+import {
+  executeMigrations,
+  filterMigrations,
+  getMigrationsToSkip,
+  getMigrationsWithDuplicateOrders,
+} from "../../../src/modules/migrations/utils/migrationUtils";
+
+const migrations = [
+  { name: "Migration1", module: { order: 1, run: async () => {} } },
+  { name: "Migration2", module: { order: 2, run: async () => {} } },
+  { name: "Migration4", module: { order: new Date("2024-03-07"), run: async () => {} } },
+  { name: "Migration3", module: { order: 3, run: async () => {} } },
+  {
+    name: "Migration5",
+    module: {
+      order: new Date("2024-04-07"),
+      run: async () => {
+        throw new Error("Test fail");
+      },
+    },
+  },
+] as const satisfies ReadonlyArray<Migration>;
+
+const createMigrationStatus = (
+  name: string,
+  order: number | Date,
+  success: boolean,
+  lastOperation: "run" | "rollback",
+): MigrationStatus => ({ name, order, time: new Date(), success, lastOperation });
+
+describe("filterMigrations", () => {
+  it("Correctly filters migrations for all parameter", () => {
+    const filteredMigrations = filterMigrations(migrations, { all: true });
+
+    expect(filteredMigrations).toStrictEqual(migrations);
+  });
+
+  it("Correctly filters migrations for range parameter", () => {
+    const filteredMigrations = filterMigrations(migrations, { range: { from: 2, to: new Date("2024-03-07") } });
+
+    expect(filteredMigrations).toStrictEqual([migrations[1], migrations[2], migrations[3]]);
```

Comment:
I know this is works as intended, but I find it quite unintuitive that the "Migration3" gets picked up as well. Maybe we should consider refusing ranges combined of a number and a date. What do you think?

Reply:
Why? I think it is nice if you had a folder of combined orders. I believe that the ordering with dates sucks, but I don;t really wanna delete, because somebody might use it :D